dockerBuild fails for Apple silliconMac

[Generate an app with the defaults. ](https://launch.micronaut.io)

```
./gradlew dockerBuild

> Task :dockerfile
Dockerfile written to: /Users/sdelamo/Downloads/dockertest/build/docker/main/Dockerfile

> Task :dockerBuild
Building image using context '/Users/sdelamo/Downloads/dockertest/build/docker/main'.
Using Dockerfile '/Users/sdelamo/Downloads/dockertest/build/docker/main/Dockerfile'
Using images 'demo'.
Step 1/8 : FROM openjdk:17-alpine

> Task :dockerBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dockerBuild'.
> Could not build image: no matching manifest for linux/arm64/v8 in the manifest list entries

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
6 actionable tasks: 6 executed
```

### Environment Information

Operating system: MacOS Ventura
Architecture: Apple Silicon Mac
Micronaut Gradle Plugin 4.0.2

### Example Application

_No response_

### Version

4.0.2

But the above is not working when executing **optimizedDockerBuild**

`Step 1/8 : FROM openjdk:17-alpine

> Task :app:optimizedDockerBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:optimizedDockerBuild'.
> Could not build image: no matching manifest for linux/arm64/v8 in the manifest list entries
`

--- a/docker-plugin/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/docker-plugin/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -23,6 +23,7 @@ import java.util.List;
 
 public abstract class MicronautDockerfile extends Dockerfile implements DockerBuildOptions {
     public static final String DEFAULT_WORKING_DIR = "/home/app";
+    public static final String DEFAULT_BASE_IMAGE = "eclipse-temurin:17-jre-focal";
 
     @Input
     private final Property<String> baseImage;
@@ -109,7 +110,7 @@ public abstract class MicronautDockerfile extends Dockerfile implements DockerBu
             case LAMBDA:
                 javaApplication.getMainClass().set("io.micronaut.function.aws.runtime.MicronautLambdaRuntime");
             default:
-                from(new Dockerfile.From(from != null ? from : "openjdk:17-alpine"));
+                from(new Dockerfile.From(from != null ? from : DEFAULT_BASE_IMAGE));
                 setupResources(this);
                 exposePort(exposedPorts);
                 getInstructions().addAll(additionalInstructions);

--- a/docker-plugin/src/test/groovy/io/micronaut/gradle/DockerBuildTaskSpec.groovy
+++ b/docker-plugin/src/test/groovy/io/micronaut/gradle/DockerBuildTaskSpec.groovy
@@ -80,7 +80,7 @@ class Application {
 """
 
         testProjectDir.newFile("Dockerfile") << """
-FROM openjdk:17-alpine
+FROM eclipse-temurin:17-jre-focal
 WORKDIR /home/alternate
 COPY layers/libs /home/alternate/libs
 COPY layers/classes /home/alternate/classes
@@ -189,7 +189,7 @@ class Application {
 
         and:
         def dockerfile = new File(testProjectDir.root, 'build/docker/main/Dockerfile').text
-        dockerfile == """FROM openjdk:17-alpine
+        dockerfile == """FROM eclipse-temurin:17-jre-focal
 WORKDIR /home/alternate
 COPY layers/libs /home/alternate/libs
 COPY layers/classes /home/alternate/classes

--- a/functional-tests/src/test/groovy/io/micronaut/gradle/aot/MicronautAOTDockerSpec.groovy
+++ b/functional-tests/src/test/groovy/io/micronaut/gradle/aot/MicronautAOTDockerSpec.groovy
@@ -20,7 +20,7 @@ class MicronautAOTDockerSpec extends AbstractAOTPluginSpec {
         result.task(":optimizedDockerfile").outcome != TaskOutcome.FAILED
 
         def dockerFile = normalizeLineEndings(file("build/docker/optimized/Dockerfile").text)
-        dockerFile == """FROM openjdk:17-alpine
+        dockerFile == """FROM eclipse-temurin:17-jre-focal
 WORKDIR /home/app
 COPY layers/libs /home/app/libs
 COPY layers/classes /home/app/classes
@@ -47,7 +47,7 @@ ENTRYPOINT ["java", "-jar", "/home/app/application.jar"]
         result.task(":optimizedDockerBuild").outcome != TaskOutcome.FAILED
 
         def dockerFile = normalizeLineEndings(file("build/docker/optimized/Dockerfile").text)
-        dockerFile == """FROM openjdk:17-alpine
+        dockerFile == """FROM eclipse-temurin:17-jre-focal
 WORKDIR /home/app
 COPY layers/libs /home/app/libs
 COPY layers/classes /home/app/classes

--- a/functional-tests/src/test/groovy/io/micronaut/gradle/docker/DockerNativeFunctionalTest.groovy
+++ b/functional-tests/src/test/groovy/io/micronaut/gradle/docker/DockerNativeFunctionalTest.groovy
@@ -689,7 +689,7 @@ class Application {
 
         then:
         def dockerfile = new File(testProjectDir.root, 'build/docker/main/Dockerfile').text
-        dockerfile == """FROM openjdk:17-alpine
+        dockerfile == """FROM eclipse-temurin:17-jre-focal
 WORKDIR /home/app
 COPY layers/libs /home/app/libs
 COPY server.iprof /home/app/server.iprof
@@ -769,7 +769,7 @@ class Application {
 
         then:
         def dockerfile = new File(testProjectDir.root, 'build/docker/main/Dockerfile').text
-        dockerfile == """FROM openjdk:17-alpine
+        dockerfile == """FROM eclipse-temurin:17-jre-focal
 WORKDIR /home/app
 COPY layers/libs /home/app/libs
 COPY server.iprof /home/app/server.iprof

--- a/src/docs/asciidoc/index.adoc
+++ b/src/docs/asciidoc/index.adoc
@@ -2,7 +2,7 @@
 :native-gradle-plugin: https://graalvm.github.io/native-build-tools/{native-build-tools-version}/gradle-plugin.html
 :gradle-docs: https://docs.gradle.org/{gradle-version}/userguide
 :gradle-toolchains: {gradle-docs}/toolchains.html
-:default-docker-image: openjdk:17-alpine
+:default-docker-image: eclipse-temurin:17-jre-focal
 :docker-plugin: https://github.com/bmuschko/gradle-docker-plugin
 :aws-docs: https://micronaut-projects.github.io/micronaut-aws/latest/guide/index.html#customRuntimes
 :testresources-guide: https://micronaut-projects.github.io/micronaut-test-resources/latest/guide

